OutOfSchool/OutOfSchool.WebApi/Controllers/V1/BlockedProviderParentController.cs

```diff
@@ -0,0 +1,124 @@
+        /// </summary>
+        /// <param name="blockedProviderParentUnblockDto">blockedProviderParentUnblockDto entity to update.</param>
+        /// <returns>A <see cref="Task{BlockedProviderParentDto}"/> representing the result of the asynchronous operation.</returns>
+        /// <response code="200">BlockedProviderParent was successfully updated.</response>
+        /// <response code="400">BlockedProviderParentUnblockDto was incorrect. Block does not exist with current ProviderId and ParentId.</response>
+        /// <response code="401">If the user is not authorized.</response>
+        /// <response code="500">If any server error occures.</response>
+        [HasPermission(Permissions.ProviderAdmins)]
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BlockedProviderParentDto))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<IActionResult> UnBlock(BlockedProviderParentUnblockDto blockedProviderParentUnblockDto)
+        {
+            var userId = GetUserId();
+            var result = await blockedProviderParentService.Unblock(blockedProviderParentUnblockDto, userId).ConfigureAwait(false);
+
+            if (!result.Succeeded)
+            {
+                return BadRequest(result.OperationResult);
+            }
+
+            return Ok(result.Value);
+        }
+
+        /// <summary>
+        /// Return blocked entity.
+        /// </summary>
+        /// <param name="parentId">Key of the Parent in table.</param>
+        /// <param name="providerId">Key of the Provider in table.</param>
+        /// <returns>A <see cref="Task{BlockedProviderParentDto}"/> that was found or null.</returns>
+        /// <response code="200">Operation was executed successfully.</response>
+        /// <response code="204">Blocked entity was not found.</response>
+        /// <response code="400">Id was wrong.</response>
+        /// <response code="401">If the user is not authorized.</response>
+        /// <response code="500">If any server error occures.</response>
+        [Authorize]
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BlockedProviderParentDto))]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<IActionResult> GetBlock(Guid parentId, Guid providerId)
+        {
+            return Ok(await blockedProviderParentService.GetBlock(parentId, providerId).ConfigureAwait(false));
+        }
+
+        private string GetUserId()
```

Comment:
i think we already have an extension for that. Please check common project or webapi extensions.